packages/calcite-components/src/components/input-date-picker/input-date-picker.e2e.ts

```diff
@@ -353,23 +353,40 @@ describe("calcite-input-date-picker", () => {
         await skipAnimations(page);
         await page.waitForChanges();
         inputDatePicker = await page.find("calcite-input-date-picker");
+        type InputDatePickerEventOrderWindow = GlobalTestProps<{ events: string[] }>;
+
+        await page.$eval("calcite-input-date-picker", (sheet: HTMLCalciteInputDatePickerElement) => {
+          const receivedEvents: string[] = [];
+          (window as InputDatePickerEventOrderWindow).events = receivedEvents;
+
+          ["calciteInputDatePickerOpen", "calciteInputDatePickerClose"].forEach((eventType) => {
+            sheet.addEventListener(eventType, (event) => receivedEvents.push(event.type));
+          });
+        });
       });
 
       it("toggles the date picker when clicked", async () => {
+        const openSpy = await inputDatePicker.spyOnEvent("calciteInputDatePickerOpen");
```

Comment:
From what I remember we introduce those in some places to avoid multiple `await page.waitForChanges();`

Reply:
I'll open an issue to add user interaction test to the helper.

Reply:
> From what I remember we introduced those in some places to avoid multiple await page.waitForChanges();

I believe we used `waitForEvent` for that. This is using event spies, which is different.